02-peer-protocol.md

```diff
@@ -2439,6 +2957,60 @@ such detection is left as an option.
 Nodes inside a blinded route must use `invalid_onion_blinding` to avoid
 leaking information to senders trying to probe the blinded route.
 
+### Batching channel messages
+
+Multiple channel messages can be grouped together as a single logical message
+using the `start_batch` message.
+
+1. type: 127 (`start_batch`)
+2. data:
+   * [`channel_id`:`channel_id`]
+   * [`u16`:`batch_size`]
+
+1. `tlv_stream`: `start_batch_tlvs`
+2. types:
+   1. type: 1 (`message_type`)
+   2. data:
+     * [`u16`:`message_type`]
+
+#### Requirements
+
+A sending node:
+  - MUST set `batch_size` to a value strictly greater than 1.
+  - MUST set `batch_size` to a value lower than or equal to 20.
+  - If the batch only contains messages of the same type:
+    - MUST set `message_type` to the corresponding
+      [Bolt 1 type](./01-messaging.md#lightning-message-format).
+  - After sending `start_batch`:
+    - MUST send `batch_size` messages for the same `channel_id`, without any
+      other unrelated messages in-between.
+    - If `message_type` is set:
+      - MUST send `batch_size` messages of the same message type, without any
+        other unrelated messages in-between.
+
+A receiving node:
+  - If `batch_size` is not strictly greater than 1:
+    - MUST ignore the `start_batch` message.
+    - SHOULD send a `warning`.
+  - If `batch_size` is strictly greater than 20:
+    - MUST send a `warning` and close the connection, or send an `error` and
+      fail the channel.
+  - MUST group the next `batch_size` messages and process them together.
+  - If one of those messages is not for the specified `channel_id`:
+    - MUST send a `warning` and close the connection, or send an `error` and
+      fail the channel.
+  - If `message_type` is set, but one of those messages is not of the specified
+    message type:
+    - MUST send a `warning` and close the connection, or send an `error` and
+      fail the channel.
```

Comment:
```
  - If `message_type` is set, but one of those messages is not of the specified
    message type:
    - MUST send a `warning` and close the connection, or send an `error` and
      fail the channel.
```

Should we change this to
```
  - if `message_type` is not present or not set to the type for `commitment_signed`
    - MUST ignore the `start_batch` message and process the following messages sequentially
```

So the behavior (for now) is consistent across implementations and later on we would need to explicitly update the spec to support more types?